src/explore-education-statistics-common/src/modules/charts/components/CustomReferenceLineLabel.tsx

```diff
@@ -0,0 +1,44 @@
+import { ReferenceLine } from '@common/modules/charts/types/chart';
+import { ChartData } from '@common/modules/charts/types/dataSet';
+import React from 'react';
+import { ReferenceLineProps } from 'recharts';
+
+interface Props {
+  chartData: ChartData[];
+  referenceLine: ReferenceLine;
+  referenceLineProps: ReferenceLineProps;
```

Comment:
Might be better to have `Props extend ReferenceLineProps` and then spread the props in?

Reply:
Why would that be better?

Reply:
Generally, it's a good idea to try and keep props as flat as possible and avoid object/array props. The main reason is that if we want to use things like the `memo` HOC (for performance reasons), we need to define a custom memoization strategy e.g. to check over the contents of the object/array props (as well as the top-level props). This leads to more code and we can't just use the default by-reference memoization strategy.

Of course, it's not always practical to flatten all the props down. For example, some components may need to use all of the different parts of a really complex object(s). That said, I'd personally say it's a good idea to try and avoid passing in more props/data than a component actually needs :wink: 

In this case, I don't think it makes too much of a difference though. Happy to leave it as-is, but something to think about for other components :slightly_smiling_face:

Reply:
Thanks, will bear this in mind in future.